### ISSUE TITLE
Parse date error in the record listing module

### ISSUE THREAD

**Issue by septentrio** [contao/core#8173]
Hello,
There's a parsing problem when i want to get a date field from my DB (for exemple the date field from the 'comments' table). The date is a timestamp and is not converted to a human friendly display.

I think it should be directy converted to the date format configuration set in the website root

**Comment by discordier** [comment]
How, where and when do you expect it to get parsed.
Show us some code or at least tell us some details what he problem is.
1. What (exactly) are you doing.
2. What do you expect to happen.
3. What happens instead.

**Comment by septentrio** [comment]
Well, i want to display the date field from the tl_comments table : 

![image](https://cloud.githubusercontent.com/assets/870692/12036077/017de96e-ae43-11e5-8593-f0279c5fdaf4.png)

so i entered the date in "Fields".

![image](https://cloud.githubusercontent.com/assets/870692/12036154/9fd21180-ae43-11e5-9544-69d17ea743a7.png)

As you can see the problem is that the date is rendered as timestamp
I simply want to display it in the format d.m.Y

Thanks

**Comment by asaage** [comment]
You will need to parse this in a custom List-Template...

probably a Modification in your "Fields" field can do the same (not shure if thats acceptet by the module)

instead of ```name,date,comment```
you could try ```name, FROM_UNIX_TIMESTAMP(date, '%Y %D %M %h:%i:%s') AS date, comment```

**Comment by septentrio** [comment]
@asaage your solution doesn't work and sends an error.

I don't understand why it doesn't work, as the controller recognizes the format to apply according to the name of the field

ModuleListing.php - line 443

> // Date
		elseif ($GLOBALS['TL_DCA'][$this->list_table]['fields'][$k]['eval']['rgxp'] == 'date')
		{
			$value = \Date::parse($objPage->dateFormat, $value);
		}

**Comment by asaage** [comment]
sorry i had a typo
FROM_UNIX_TIME
 - not
FROM_UNIX_TIMESTAMP

**Comment by septentrio** [comment]
No, it doesn't work too

**Comment by asaage** [comment]
Hmm i did not know about the format-smartness of the ModuleListing.php...
Doublecheck if your rootPage has a dateFormat or if it's only set in Systemsettings!
It might be indeed a bug.

**Comment by Aybee** [comment]
Its because the field tl_comments.date is not set to rgxp=date as it is an internal date value which needs no input validation.

You can set this field to
```php
$GLOBALS['TL_DCA']['tl_comments']['fields']['date']['eval']['rgxp'] = 'date';
```
then it will work.

**Comment by septentrio** [comment]
@Aybee 

Your first solution worked partially : it displays the right date but it stripped the other fields (name and comment) :(

I havve no iea how to implement your second solution, can you help me ?

**Comment by asaage** [comment]
my solution will work - i noticed another typo: sorry for that.
i knew, because i have done that before (with the tl_log table).

Fields:
```name,FROM_UNIXTIME(date, '%d.%m.%Y') AS date,comment``` will do it

Althoug i cannot understand why the DCA-rgxp=date solution should kill the other columns....

Aybee's second solution (my very-first-one) requires you to create a 'list_default' derivative under Layout->Templates and do the magic inside the Template. a ```<?php print_r($this); ?>``` might help if you want to go that way.

**Comment by Aybee** [comment]
> Your first solution worked partially : it displays the right date but it stripped the other fields (name and comment)

Don't know why this shouldn't work. Did you put this line into your dcaconfig.php like I did?

> I have no idea how to implement your second solution, can you help me?

In your copy of list_default.html5 you have to formate the date
```php
<?php foreach ($row as $col):
  if($col['field'] == 'date')
  {
    $col['content'] = \Date::parse('d.m.Y', $col['content']);
  }
?>
```
I'm not as good in mySQL, but if the solution by @asaage works, you also can use this one. It does the formating before sending to PHP.

**Comment by septentrio** [comment]
Thanks guys for your precious help,

Finally i got it with @Aybee 's template implement.

Shouldn't it be marked as a bug to fix ?

**Comment by Aybee** [comment]
I think no as there is no other way to identify a field as a field that contains a unix timestamp. And setting rgxp=date is not logical as this field needs no input validation. Also I think that the listing template in almost every case has to be modified.

But maybe @leofeyer will set rgxp=date if he finds no bad side effect.

**Comment by leofeyer** [comment]
Indeed the `rgxp` flag is missing. However, it should be set to `datim` by default, shouldn't it?

**Comment by Aybee** [comment]
Don't know. Everybody wants a different date format in FE anyway. And there's no input for this field, so why an input validation?

Differs the output from setting it to 'datim' vs 'date'?

**Comment by leofeyer** [comment]
* `date` bedeutet "Datum": DD.MM.YYYY
* `datim` bedeutet "Datum und Uhrzeit": DD.MM.YYYY HH:ii

Da es bei einem Kommentar auch auf die Uhrzeit ankommt, würde ich wenn letzteres als Standardformat setzen.

**Comment by Aybee** [comment]
It's not my ticket, but seems to be ok for me.

**Comment by leofeyer** [comment]
Fixed in 04ec47ae70f55388da89f3ad24941aefbbcb5b12.
